server/py/framework/db/sqldb/db.py

```diff
@@ -5927,7 +5937,30 @@ def store_alert_activation(
             data=extra_data,
         )
 
-        self._upsert(session, [alert_activation_record])
+        # if reset_policy is MANUAL, we need to keep id to be able to update number_of_events
+        # when policy is reset
+        if alert_data.reset_policy == mlrun.common.schemas.alert.ResetPolicy.MANUAL:
+            return self._upsert_object_and_flush_to_get_field(
+                session, alert_activation_record, "id"
+            )
+        else:
+            return self._upsert(session, [alert_activation_record])
+
+    def update_alert_activation(
+        self,
+        session,
+        activation_id: int,
+        activation_time: datetime,
```

Comment:
Why is the activation time required if we have the record id?

Reply:
@TomerShor that's a good question. And it's indeed done on purpose. The thing is that in our case we have composite PK (id, activation_time) and if we don't pass activation time, MySQL will always go through all the partitions (even if id is passed). So we need it to improve performance.